server/webhook/merge_request.go

```diff
@@ -148,3 +231,72 @@ func (w *webhook) handleChannelMergeRequest(ctx context.Context, event *gitlab.M
 
 	return res, nil
 }
+
+func (w *webhook) calculateUserDiffs(authorID int, previousUsers, currentUsers []*gitlab.EventUser) (bool, []string, []string) {
+	mapPreviousUsers := map[int]*gitlab.EventUser{}
+	mapCurrentUsers := map[int]*gitlab.EventUser{}
+	updatedPreviousUsers := []string{}
+	updatedCurrentUsers := []string{}
+	isAuthorPresent := false
+	for _, previousUser := range previousUsers {
+		mapPreviousUsers[previousUser.ID] = previousUser
+	}
+
+	for _, currentUser := range currentUsers {
+		mapCurrentUsers[currentUser.ID] = currentUser
+		if _, exist := mapPreviousUsers[currentUser.ID]; !exist {
+			if currentUser.ID == authorID {
+				isAuthorPresent = true
+			}
+			updatedCurrentUsers = append(updatedCurrentUsers, w.gitlabRetreiver.GetUsernameByID(currentUser.ID))
+		}
+	}
+
+	for _, previousUser := range previousUsers {
+		if _, exist := mapCurrentUsers[previousUser.ID]; !exist {
+			if previousUser.ID == authorID {
+				isAuthorPresent = true
+			}
+			updatedPreviousUsers = append(updatedPreviousUsers, w.gitlabRetreiver.GetUsernameByID(previousUser.ID))
+		}
+	}
+
+	return isAuthorPresent, updatedPreviousUsers, updatedCurrentUsers
+}
+
+func (w *webhook) checkForMultipleEvents(event *gitlab.MergeEvent) bool {
+	eventCount := 0
+	if event.Changes.Labels.Current != nil || event.Changes.Labels.Previous != nil {
+		eventCount++
```

Comment:
Can the code simply return with `return true` in this case?

Reply:
Removed the above function in the updated logic as there was no need for it.